### PR TITLE
Immediately return from http read loop if upstream round-trip fails

### PR DESCRIPTION
**Pull request by oxtoacart**
For getlantern/lantern-internal#2110

### CODE REVIEW

File: proxy_http.go [hunk 1]
```diff
@@ -174,6 +181,11 @@ func (proxy *proxy) processRequests(ctx filters.Context, remoteAddr string, req
 			}
 		}
 
+		if err != nil {
+			// We encountered an error on round-tripping, stop now
+			return err
+		}
```
Comment (oxtoacart) [inline proxy_http.go]: I think the race condition must have been happening when our round-trip upstream times out. When that happens, the Transport is still trying to read from downstream to copy to upstream. But our read loop has already proceeded to the next call to `readRequest`.

The solution is to immediately return the err and not attempt to read any more.

**Comment by joesis** [comment]
LGTM. It's a correction of the proxying logic regardless. Nice catch @oxtoacart.

**Comment by joesis** [comment]
Updated dependency in getlantern/http-proxy-lantern@91c12d17ecd5b35794d07abfc41bc11e51e545d3

**Comment by oxtoacart** [comment]
Thanks @joesis !

**Pull request merged**